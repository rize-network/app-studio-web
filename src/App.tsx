import React, { Suspense } from 'react';
import { Route, Routes, useLocation } from 'react-router-dom';
import { ComponentRouter } from 'src/routes/routes';
import { View, Text } from 'app-studio';

export const App: React.FC = () => {
  const location = useLocation();
  return (
    <Suspense
      fallback={
        <View
          height="100vh"
          display="flex"
          alignItems="center"
          justifyContent="center"
        >
          <Text>Loading...</Text>
        </View>
      }
    >
      <Routes location={location}>
        {ComponentRouter.map((item, index) => (
          <Route key={index} path={item.path} element={item.element} />
        ))}
      </Routes>
    </Suspense>
  );
};

export default App;
