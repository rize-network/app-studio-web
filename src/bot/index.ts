import { Bot } from './Bot';
import { DocuCode } from './DocuCode';

async function main() {
  const projectName = 'MDX Documentation';
  const descriptionPath = 'docs/description.md';

  // Get componentName from command line arguments
  const componentName = process.argv[2]; // Assuming it's the first argument passed
  if (!componentName) {
    console.error('Please provide a component name as an argument.');
    process.exit(1);
  }

  console.log(`Generating documentation for component: ${componentName}`);

  const basePath = 'src/components';
  const propsPath = `src/data/props/${componentName}.props.json`;
  const componentFolder = `${basePath}/${componentName}`;
  const componentPath = `${basePath}/${componentName}/${componentName}`;

  const assistantGPT = new Bot(componentName);

  // Implementation for Documentation of the code
  console.log('Initializing documentation process...');
  const docuCode = new DocuCode(componentName);
  const assistantDocumentation = await assistantGPT.init(
    projectName,
    [],
    descriptionPath,
    'docu'
  );

  await docuCode.processDirectory(componentFolder, assistantDocumentation.id);

  // Generating props file for code
  console.log('Generating props file...');
  const { fileIds } = await assistantGPT.addFiles(componentPath);

  const assistantCreation = await assistantGPT.init(
    projectName,
    fileIds,
    descriptionPath,
    'props'
  );

  await assistantGPT.response(assistantCreation.id, propsPath);

  await assistantGPT.MarkdownGeneration(
    componentFolder,
    componentName,
    propsPath
  );

  console.log('Documentation generation completed.');
}

main().catch((error) => {
  console.error('Error during the execution of main:', error);
});
