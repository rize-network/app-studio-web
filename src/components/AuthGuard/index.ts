export { AuthGuard, withAuthGuard, RouteAuthGuard, AuthStatus } from './AuthGuard';
