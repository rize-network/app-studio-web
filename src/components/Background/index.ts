export { Background } from './Background';
export type {
  BackgroundProps,
  AuroraBackgroundProps,
  MeteorsProps,
  WallProps,
  ParticlesProps,
  GridProps,
  RipplesProps,
  BackgroundImageProps,
  BackgroundVideoProps,
  BackgroundGradientProps,
  BackgroundStyles,
} from './Background/Background.props';
export type {
  BackgroundEffect,
  AuroraDirection,
  MeteorTiming,
  BorderAnimationStyles,
  StrokeAnimationStyles,
  BackgroundContextType,
  BackgroundSize,
  BackgroundPosition,
  BackgroundRepeat,
  BlendMode,
} from './Background/Background.type';
