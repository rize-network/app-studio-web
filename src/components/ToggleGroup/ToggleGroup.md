# ToggleGroup

Enables selection from a group of toggle switches.

### **Import**
  ```tsx static
  import { ToggleGroup } from '@app-studio/web';
  ```

### **Default**
```tsx
import React from 'react';
import { ToggleGroup } from '../ToggleGroup';

export const DefaultDemo = () => {
  const items = [
    { id: 'B', value: 'B' },
    { id: 'C', value: 'C' },
    { id: 'D', value: 'D' },
  ];
  return <ToggleGroup items={items} />;
};

```

### **shape**
"Optional shape prop to define the shape of toggle elements."

```tsx
import React from 'react';
import { DustBinSvg } from '../../Svg';
import { ToggleGroup } from '../ToggleGroup';
import { Shape } from '../ToggleGroup/ToggleGroup.type';
import { Horizontal } from '../../Layout/Horizontal/Horizontal';
import { View } from '../../Layout/View/View';

export const ShapeDemo = () => {
  const items = [
    {
      id: 'item',
      value: <DustBinSvg color="black" size={18} />,
      isActive: true,
    },
  ];
  return (
    <Horizontal gap={15}>
      {['sharp', 'rounded', 'pillShaped'].map((border, index) => (
        <View position="relative" key={index}>
          <ToggleGroup items={items} shape={border as Shape} />
        </View>
      ))}
    </Horizontal>
  );
};

```

### **items**
"Mandatory array of items defining each toggle in the group."

```tsx
import React from 'react';
import { ToggleGroup } from '../ToggleGroup';
import { Text } from '../../Text/Text';

export const ItemsDemo = () => {
  const items = [
    { id: 'B', value: <Text>B</Text> },
    { id: 'C', value: <Text>C</Text> },
    { id: 'D', value: <Text>D</Text> },
  ];
  return <ToggleGroup items={items} />;
};

```

### **variant**
"Optional variant prop to specify the visual style of the toggle group."

```tsx
import React from 'react';
import { Text } from '../../Text/Text';
import { ToggleGroup } from '../ToggleGroup';
import { Variant } from '../ToggleGroup/ToggleGroup.type';
import { Vertical } from '../../Layout/Vertical/Vertical';

export const VariantDemo = () => {
  const items = [
    { id: 'B', value: <Text>B</Text> },
    { id: 'C', value: <Text>C</Text> },
    { id: 'D', value: <Text>D</Text> },
  ];
  return (
    <Vertical gap={15}>
      {['outline', 'link', 'ghost'].map((variant, index) => (
        <ToggleGroup key={index} items={items} variant={variant as Variant} />
      ))}
    </Vertical>
  );
};

```

### **colorScheme**
"Optional colorScheme prop for theming the toggle group."

```tsx
import React from 'react';
import { ToggleGroup } from '../ToggleGroup';
import { Vertical } from '../../Layout/Vertical/Vertical';
import { Text } from '../../Text/Text';

export const ColorSchemeDemo = () => {
  const items = [
    { id: 'B', value: <Text>B</Text> },
    { id: 'C', value: <Text>C</Text>, isActive: true },
    { id: 'D', value: <Text>D</Text>, isDisabled: true },
  ];
  return (
    <Vertical gap={10}>
      {[
        'theme.primary',
        'theme.secondary',
        'theme.warning',
        'theme.success',
        'theme.error',
      ].map((color, index) => (
        <ToggleGroup key={index} items={items} colorScheme={color} />
      ))}
    </Vertical>
  );
};

```

### **onToggleChange**
"Optional callback function that fires when the active toggle changes, providing the IDs of active toggles."

```tsx
import React, { useState } from 'react';
import { ToggleGroup } from '../ToggleGroup';
import { Text } from '../../Text/Text';
import { Vertical } from '../../Layout/Vertical/Vertical';

export const OnToggleChangeDemo = () => {
  const [newItems, setNewItems] = useState<any>([]);

  const items = [
    { id: 'B', value: <Text>B</Text> },
    { id: 'C', value: <Text>C</Text> },
    { id: 'D', value: <Text>D</Text> },
  ];
  return (
    <Vertical gap={10}>
      <ToggleGroup
        items={items}
        onToggleChange={(items) => setNewItems(items)}
      />
      Selected: {newItems}
    </Vertical>
  );
};

```

