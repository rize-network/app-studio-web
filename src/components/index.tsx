export * from './Button/Button';
export * from './Form/Checkbox/Checkbox';
export * from './Form/CountryPicker/CountryPicker';
export * from './Form/DatePicker/DatePicker';
export * from './Form/Password/Password';
export * from './Form/Select/Select';
export * from './Form/Switch/Switch';
export * from './Form/Switch/Switch';
export * from './Form/TextArea/TextArea';
export * from './Form/TextField/TextField';
export * from './Layout/Center/Center';
export * from './Formik';
export * from './Layout/Horizontal/Horizontal';
export * from './Layout/Vertical/Vertical';
export * from './Layout/View/View';
export * from './Link/Link';
export * from './Loader/Loader';
export * from './Modal/Modal';
export * from './Text/Text';
export * from './File/File';
export * from './Tabs/Tabs';
export * from './Alert/Alert';
export * from './AspectRatio/AspectRatio';
export * from './Avatar/Avatar';
export * from './Badge/Badge';
export * from './Toggle/Toggle';
export * from './ToggleGroup/ToggleGroup';
export * from './Message/Message';
