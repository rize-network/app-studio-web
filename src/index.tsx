import React from 'react';
import ReactDOM from 'react-dom/client';
import AppProvider from 'src/providers';
import { App } from './App';
import { ComboBoxProvider } from './components/Form/ComboBox/ComboBox/ComboBox.provider';

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);
root.render(
  <React.StrictMode>
    <AppProvider>
      <ComboBoxProvider>
        <App />
      </ComboBoxProvider>
    </AppProvider>
  </React.StrictMode>
);
