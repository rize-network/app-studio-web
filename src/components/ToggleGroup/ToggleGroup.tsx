import React from 'react';
import { ToggleGroupProps } from './ToggleGroup/ToggleGroup.props';
import { useToggleGroupState } from './ToggleGroup/ToggleGroup.state';
import { ToggleGroupView } from './ToggleGroup/ToggleGroup.view';

const ToggleGroupComponent = ({
  items,
  shape,
  colorScheme,
  variant,
  onToggleChange,
}: ToggleGroupProps) => {
  const { activeToggles, setActiveToggles } = useToggleGroupState();

  return (
    <ToggleGroupView
      items={items}
      shape={shape}
      colorScheme={colorScheme}
      variant={variant}
      activeToggles={activeToggles}
      setActiveToggles={setActiveToggles}
      onToggleChange={onToggleChange}
    />
  );
};

export const ToggleGroup = ToggleGroupComponent;
