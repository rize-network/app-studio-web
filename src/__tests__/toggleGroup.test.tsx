// Import necessary libraries and components
import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render, screen } from '@testing-library/react';
import { Shape } from 'src/components/ToggleGroup/ToggleGroup/ToggleGroup.type';
import { ToggleGroup } from 'src/components/ToggleGroup/ToggleGroup';
import { Text } from 'src/components';

// Additional info (mocked for testing if not imported)
export const ToggleGroupShapes: Record<Shape, number | string> = {
  sharp: '0px',
  rounded: '4px',
  pillShaped: '24px',
};

const items = [
  { id: 'B', value: <Text>B</Text> },
  { id: 'C', value: <Text>C</Text>, isActive: true },
  { id: 'D', value: <Text>D</Text>, isDisabled: true },
];

// Test setup and teardown
afterEach(cleanup);

// Test cases
describe('ToggleGroup Component', () => {
  // Basic Rendering
  test('renders ToggleGroup component', () => {
    render(<ToggleGroup items={items} />);
    const ToggleGroupElement = screen.getByRole('ToggleGroup');
    expect(ToggleGroupElement).toBeInTheDocument();
  });

  test('ToggleGroup to match snapshot', () => {
    const tree = renderer.create(<ToggleGroup items={items} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
