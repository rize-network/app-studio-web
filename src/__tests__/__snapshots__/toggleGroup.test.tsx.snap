// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleGroup Component ToggleGroup to match snapshot 1`] = `
<div
  className="sc-bdfCDU gxaCay"
  display="flex"
  role="ToggleGroup"
>
  <div
    className="sc-bdfCDU dmlQCl"
    color="color.trueGray.400"
    cursor="pointer"
    display="flex"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="toggle-B"
    width="fit-content"
  >
    <div
      className="sc-bdfCDU FxXNY"
      fontSize={14}
      fontStyle="normal"
      fontWeight={400}
      role="text"
      textDecoration="none"
    >
      B
    </div>
  </div>
  <div
    className="sc-bdfCDU eTueVD"
    color="color.white"
    cursor="pointer"
    display="flex"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="toggle-C"
    width="fit-content"
  >
    <div
      className="sc-bdfCDU FxXNY"
      fontSize={14}
      fontStyle="normal"
      fontWeight={400}
      role="text"
      textDecoration="none"
    >
      C
    </div>
  </div>
  <div
    className="sc-bdfCDU icIFMv"
    color="theme.disabled"
    cursor="not-allowed"
    display="flex"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="toggle-D"
    width="fit-content"
  >
    <div
      className="sc-bdfCDU FxXNY"
      fontSize={14}
      fontStyle="normal"
      fontWeight={400}
      role="text"
      textDecoration="none"
    >
      D
    </div>
  </div>
</div>
`;
